app/services/create_approval_request.rb

```diff
@@ -0,0 +1,51 @@
+class CreateApprovalRequest < ApprovalServiceApi
+  def process
+    order.order_items.each do |order_item|
+      submit_approval_request(order_item)
+    end
+    order.update(:state => 'Ordered', :ordered_at => DateTime.now.utc)
+    order
+  rescue StandardError => e
+    Rails.logger.error("CreateApprovalRequest #{e.message}")
+    raise
+  end
+
+  private
+
+  def submit_approval_request(order_item)
+    pf_item = portfolio_item(order_item)
+    body = request_body(pf_item, order_item)
+    begin
+      request = api_instance.add_request(pf_item.approval_workflow_ref, body)
+      order_item.approval_request_ref = request.id
+      order_item.save!
+    rescue ApprovalAPIClient::ApiError => e
+      Rails.logger.error("UsersApi->add_request #{e.message}")
+      raise
+    end
+  end
+
+  def order
+    @order ||= Order.find_by!(:id => params[:order_id])
+  end
+
+  def portfolio_item(order_item)
+    PortfolioItem.find_by!(:id => order_item.portfolio_item_id)
+  end
+
+  def request_body(pf_item, order_item)
+    o_params = ActionController::Parameters.new('order_item_id' => order_item.id)
+    content = OrderItemSanitizedParameters.new(o_params).process
+    ApprovalAPIClient::Request.new(
+      'requester' => username,
+      'name'      => pf_item.name,
+      'content'   => content.to_json
```

Comment:
do we want to define the content with a schema?

Reply:
There is already a schema that is defined for the plan, this content is built from that schema

Reply:
The parameter schema for service portal may not be the same as the one for approval service. It may be a simplified version by removing some less important information.

Reply:
We need content to be in json object format rather than string. @hsong-rh please verify.

Reply:
@bzwei No, content.to_json is the right format for client.

Reply:
I looked the approval api spec. `content` is a string, but the column in database is a jsonb. There is a mismatch here.
@hsong-rh we should change the swagger spec.